src/components/TextField.svelte

```diff
@@ -0,0 +1,116 @@
+<script lang="ts">
+    import { onMount } from 'svelte';
+    import { fade } from 'svelte/transition';
+
+    export let textFieldText: string;
+    export let errorCriteria: (val: string) => boolean = () => true; // Default to a function that always returns true
+    export let errorText = 'Error'; // Default error text
+
+    let value = '';
+    let isFocused = false;
+    let isValid = true;
+
+    const emptyState =
+        'w-602px h-68px font-suisse text-18px text-left bg-transparent border border-gray-600 text-white focus:outline-none transition duration-500 ease-in-out rounded-lg';
+    const focusedState =
+        'w-602px h-68px font-suisse text-18px text-left bg-transparent border border-purple-500 text-white focus:outline-none transition duration-500 ease-in-out rounded-lg';
+    const filledState =
+        'w-602px h-68px font-suisse text-18px text-left bg-transparent border border-red-500 text-white focus:outline-none transition duration-500 ease-in-out rounded-lg';
+
+    onMount(() => {
+        if (value) isFocused = true;
+    });
+
+    function handlePaste() {
+        navigator.clipboard.readText().then((clipText) => {
+            value = clipText;
+            isFocused = false;
+            isValid = errorCriteria(value);
+        });
+    }
+
+    function handleBlur() {
+        isFocused = false;
+        isValid = errorCriteria(value);
+    }
+</script>
+
+<div
+    class="relative h-68px w-602px absolute top-213px left-20px p-4 rounded-lg flex items-center gap-273px"
+>
+    <input
+        bind:value
+        type="text"
+        placeholder={textFieldText}
+        on:focus={() => (isFocused = true)}
+        on:blur={handleBlur}
+        class={value === ''
+            ? emptyState
+            : isFocused
+            ? focusedState
+            : isValid
+            ? emptyState
+            : filledState}
```

Comment:
Prettier did this automatically,
it's really ugly imo

Reply:
If you're doing ternaries this long, you might want to consider making a function that returns a string of styles.